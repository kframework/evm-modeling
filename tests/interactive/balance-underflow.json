mkAcct 0
load "account" : {
    0 : { "balance" : 1000000000
        , "code"    : "0x606060405260043610610041576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff16806396f40e3d14610046575b600080fd5b341561005157600080fd5b6100de600480803590602001908201803590602001908080602002602001604051908101604052809392919081815260200183836020028082843782019150505050505091908035906020019082018035906020019080806020026020016040519081016040528093929190818152602001838360200280828437820191505050505050919050506100e0565b005b81600090805190602001906100f6929190610112565b50806001908051906020019061010d929190610112565b505050565b82805482825590600052602060002090810192821561014e579160200282015b8281111561014d578251825591602001919060010190610132565b5b50905061015b919061015f565b5090565b61018191905b8082111561017d576000816000905550600101610165565b5090565b905600a165627a7a7230582066ac47e53c1786d7d8eb7962703666874d0636ba759b52c63d277c261372a96a0029"
        , "nonce"   : 0
        , "storage" : { }
        }
}

mkTX 0
load "transaction" : {
    0 : { "data"     : ""
        , "gasLimit" : 10000000000
        , "gasPrice" : 1
        , "nonce"    : 0
        , "r"        : "0x9bae3efe7d9d16200582021c8f7c93c143c3fb1dd6be8a83213f3a5f1ad03926"
        , "s"        : "0x177d82a90f883a19de8c6d62ce79c753ae4ea076de047ca19a017879a1ce8bc1"
        , "to"       : 0
        , "v"        : "0x1b"
        , "value"    : 200
        }
}

loadTx(0)
exception
failure "should throw exception for balance underflow"
success

.EthereumSimulation
