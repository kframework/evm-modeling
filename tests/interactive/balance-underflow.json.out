<generatedTop>
  <k>
    .
  </k>
  <exit-code>
    0
  </exit-code>
  <mode>
    SUCCESS
  </mode>
  <schedule>
    DEFAULT
  </schedule>
  <analysis>
    .Map
  </analysis>
  <ethereum>
    <evm>
      <output>
        .WordStack
      </output>
      <memoryUsed>
        0
      </memoryUsed>
      <callDepth>
        -1
      </callDepth>
      <callStack>
        .List
      </callStack>
      <interimStates>
        .List
      </interimStates>
      <substateStack>
        .List
      </substateStack>
      <callLog>
        .Set
      </callLog>
      <touchedAccounts>
        SetItem ( 0 )
      </touchedAccounts>
      <txExecState>
        <program>
          .Map
        </program>
        <programBytes>
          .WordStack
        </programBytes>
        <id>
          0
        </id>
        <caller>
          0
        </caller>
        <callData>
          .WordStack
        </callData>
        <callValue>
          0
        </callValue>
        <wordStack>
          .WordStack
        </wordStack>
        <localMem>
          .Map
        </localMem>
        <pc>
          0
        </pc>
        <gas>
          0
        </gas>
        <previousGas>
          0
        </previousGas>
        <static>
          false
        </static>
      </txExecState>
      <substate>
        <selfDestruct>
          .Set
        </selfDestruct>
        <log>
          .List
        </log>
        <refund>
          0
        </refund>
      </substate>
      <gasPrice>
        1
      </gasPrice>
      <origin>
        0
      </origin>
      <previousHash>
        0
      </previousHash>
      <ommersHash>
        0
      </ommersHash>
      <coinbase>
        0
      </coinbase>
      <stateRoot>
        0
      </stateRoot>
      <transactionsRoot>
        0
      </transactionsRoot>
      <receiptsRoot>
        0
      </receiptsRoot>
      <logsBloom>
        .WordStack
      </logsBloom>
      <difficulty>
        0
      </difficulty>
      <number>
        0
      </number>
      <gasLimit>
        0
      </gasLimit>
      <gasUsed>
        0
      </gasUsed>
      <timestamp>
        0
      </timestamp>
      <extraData>
        .WordStack
      </extraData>
      <mixHash>
        0
      </mixHash>
      <blockNonce>
        0
      </blockNonce>
      <ommerBlockHeaders>
        [ .JSONList ]
      </ommerBlockHeaders>
      <blockhash>
        .List
      </blockhash>
    </evm>
    <network>
      <activeAccounts>
        SetItem ( 0 )
      </activeAccounts>
      <accounts>
        AccountCellMapItem ( <acctID>
          0
        </acctID> , <account>
          <acctID>
            0
          </acctID>
          <balance>
            5000000001000000000
          </balance>
          <code>
            96 : 96 : 96 : 64 : 82 : 96 : 4 : 54 : 16 : 97 : 0 : 65 : 87 : 96 : 0 : 53 : 124 : 1 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 0 : 144 : 4 : 99 : 255 : 255 : 255 : 255 : 22 : 128 : 99 : 150 : 244 : 14 : 61 : 20 : 97 : 0 : 70 : 87 : 91 : 96 : 0 : 128 : 253 : 91 : 52 : 21 : 97 : 0 : 81 : 87 : 96 : 0 : 128 : 253 : 91 : 97 : 0 : 222 : 96 : 4 : 128 : 128 : 53 : 144 : 96 : 32 : 1 : 144 : 130 : 1 : 128 : 53 : 144 : 96 : 32 : 1 : 144 : 128 : 128 : 96 : 32 : 2 : 96 : 32 : 1 : 96 : 64 : 81 : 144 : 129 : 1 : 96 : 64 : 82 : 128 : 147 : 146 : 145 : 144 : 129 : 129 : 82 : 96 : 32 : 1 : 131 : 131 : 96 : 32 : 2 : 128 : 130 : 132 : 55 : 130 : 1 : 145 : 80 : 80 : 80 : 80 : 80 : 80 : 145 : 144 : 128 : 53 : 144 : 96 : 32 : 1 : 144 : 130 : 1 : 128 : 53 : 144 : 96 : 32 : 1 : 144 : 128 : 128 : 96 : 32 : 2 : 96 : 32 : 1 : 96 : 64 : 81 : 144 : 129 : 1 : 96 : 64 : 82 : 128 : 147 : 146 : 145 : 144 : 129 : 129 : 82 : 96 : 32 : 1 : 131 : 131 : 96 : 32 : 2 : 128 : 130 : 132 : 55 : 130 : 1 : 145 : 80 : 80 : 80 : 80 : 80 : 80 : 145 : 144 : 80 : 80 : 97 : 0 : 224 : 86 : 91 : 0 : 91 : 129 : 96 : 0 : 144 : 128 : 81 : 144 : 96 : 32 : 1 : 144 : 97 : 0 : 246 : 146 : 145 : 144 : 97 : 1 : 18 : 86 : 91 : 80 : 128 : 96 : 1 : 144 : 128 : 81 : 144 : 96 : 32 : 1 : 144 : 97 : 1 : 13 : 146 : 145 : 144 : 97 : 1 : 18 : 86 : 91 : 80 : 80 : 80 : 86 : 91 : 130 : 128 : 84 : 130 : 130 : 85 : 144 : 96 : 0 : 82 : 96 : 32 : 96 : 0 : 32 : 144 : 129 : 1 : 146 : 130 : 21 : 97 : 1 : 78 : 87 : 145 : 96 : 32 : 2 : 130 : 1 : 91 : 130 : 129 : 17 : 21 : 97 : 1 : 77 : 87 : 130 : 81 : 130 : 85 : 145 : 96 : 32 : 1 : 145 : 144 : 96 : 1 : 1 : 144 : 97 : 1 : 50 : 86 : 91 : 91 : 80 : 144 : 80 : 97 : 1 : 91 : 145 : 144 : 97 : 1 : 95 : 86 : 91 : 80 : 144 : 86 : 91 : 97 : 1 : 129 : 145 : 144 : 91 : 128 : 130 : 17 : 21 : 97 : 1 : 125 : 87 : 96 : 0 : 129 : 96 : 0 : 144 : 85 : 80 : 96 : 1 : 1 : 97 : 1 : 101 : 86 : 91 : 80 : 144 : 86 : 91 : 144 : 86 : 0 : 161 : 101 : 98 : 122 : 122 : 114 : 48 : 88 : 32 : 102 : 172 : 71 : 229 : 60 : 23 : 134 : 215 : 216 : 235 : 121 : 98 : 112 : 54 : 102 : 135 : 77 : 6 : 54 : 186 : 117 : 155 : 82 : 198 : 61 : 39 : 124 : 38 : 19 : 114 : 169 : 106 : 0 : 41 : .WordStack
          </code>
          <storage>
            .Map
          </storage>
          <nonce>
            1
          </nonce>
        </account> )
      </accounts>
      <txOrder>
        ListItem ( 0 )
      </txOrder>
      <txPending>
        .List
      </txPending>
      <messages>
        MessageCellMapItem ( <msgID>
          0
        </msgID> , <message>
          <msgID>
            0
          </msgID>
          <txNonce>
            0
          </txNonce>
          <txGasPrice>
            1
          </txGasPrice>
          <txGasLimit>
            10000000000
          </txGasLimit>
          <to>
            0
          </to>
          <value>
            200
          </value>
          <v>
            813183330
          </v>
          <r>
            48 : 120 : 57 : 98 : 97 : 101 : 51 : 101 : 102 : 101 : 55 : 100 : 57 : 100 : 49 : 54 : 50 : 48 : 48 : 53 : 56 : 50 : 48 : 50 : 49 : 99 : 56 : 102 : 55 : 99 : 57 : 51 : 99 : 49 : 52 : 51 : 99 : 51 : 102 : 98 : 49 : 100 : 100 : 54 : 98 : 101 : 56 : 97 : 56 : 51 : 50 : 49 : 51 : 102 : 51 : 97 : 53 : 102 : 49 : 97 : 100 : 48 : 51 : 57 : 50 : 54 : .WordStack
          </r>
          <s>
            48 : 120 : 49 : 55 : 55 : 100 : 56 : 50 : 97 : 57 : 48 : 102 : 56 : 56 : 51 : 97 : 49 : 57 : 100 : 101 : 56 : 99 : 54 : 100 : 54 : 50 : 99 : 101 : 55 : 57 : 99 : 55 : 53 : 51 : 97 : 101 : 52 : 101 : 97 : 48 : 55 : 54 : 100 : 101 : 48 : 52 : 55 : 99 : 97 : 49 : 57 : 97 : 48 : 49 : 55 : 56 : 55 : 57 : 97 : 49 : 99 : 101 : 56 : 98 : 99 : 49 : .WordStack
          </s>
          <data>
            .WordStack
          </data>
        </message> )
      </messages>
    </network>
  </ethereum>
</generatedTop>
